### PR TITLE
Feature to add permission for user to manage their own integrations only

### LINKED ISSUE #2900: Create role to manage own integrations only
Your Rocket.Chat version: 0.26.0

As an administrator, I'd like to grant right of administration over a user's own created integrations, ensuring they don't interfere with anybody else's. This means creation of a new role, such as 'manage-own-integrations', which limits the user to the scope of their own integrations in terms of viewing, modifying and deleting.

### PR DESCRIPTION
**Pull request by lochiiconnectivity**
@RocketChat/core 

Closes #2900 

A new permission is added 'manage-own-integrations'.

The new permission, if present without 'manage-integrations', grants
the user access to add integrations, but only to view, modify or delete their own.

This permission is given to the 'bot' role , which has 'manage-integrations'
removed from it.

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
@lochiiconnectivity thanks for the PR.  

One thing we need to make sure of here... We need to make sure that a user cannot add an integration for a room they are not apart of.

Otherwise i'm seeing a super easy way for people to be able to eavesdrop on other rooms via an integration

**Comment by lochiiconnectivity** [comment]
I'll add this in...

**Comment by engelgabriel** [comment]
@rodrigok can you please help to review this PR?

**Comment by lochiiconnectivity** [comment]
ok, I've added this, now users with 'manage-own-integrations' can not add integrations for rooms they are not part of, however users with 'manage-integrations' can still do this, which replicates the existing behaviour (and is appropriate if these users are administrators in some sense)

**Comment by engelgabriel** [comment]
@rodrigok please review this?

File: packages/rocketchat-integrations/client/startup.coffee [hunk 1]
```diff
@@ -4,4 +4,4 @@ RocketChat.AdminBox.addOption
 	href: 'admin-integrations'
 	i18nLabel: 'Integrations'
 	permissionGranted: ->
-		return RocketChat.authz.hasAllPermission('manage-integrations')
+		return (RocketChat.authz.hasAllPermission 'manage-integrations') or (RocketChat.authz.hasAllPermission 'manage-own-integrations')
```
Comment (rodrigok) [inline packages/rocketchat-integrations/client/startup.coffee]: Just a little code improvement, you can use `RocketChat.authz.hasAtLeastOnePermission(['manage-integrations', 'manage-own-integrations'])` when checking for permissions at client side.

**Comment by rodrigok** [comment]
LGTM Maybe we should wait for the reply in my code comment

File: packages/rocketchat-integrations/client/startup.coffee (hunk 1 above)
Comment (lochiiconnectivity) [inline packages/rocketchat-integrations/client/startup.coffee]: ack, I didn't know about this, I'll do that.

**Comment by engelgabriel** [comment]
@RocketChat/core can anyone solve the conflicts please? I want to merge this today.

**Comment by lochiiconnectivity** [comment]
since nobody is doing this , I'll re-base and fix conflicts. It looks like I'll need to update the error reporting as part of this conflict resolution.

**Pull request merged**